From 963b897f06304b7af65dafbeabf3774255098aed Mon Sep 17 00:00:00 2001
From: Thomas Roos <throos@amazon.de>
Date: Wed, 21 Jan 2026 15:31:55 +0000
Subject: [PATCH] Fix cross-compilation for aarch64

Detect cross-compilation target from OECORE_TARGET_ARCH environment variable
when building for aarch64 on non-aarch64 build hosts.

Upstream-Status: Inappropriate [embedded specific]
---
 setup.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/setup.py b/setup.py
index c5691f9..e8e7e62 100644
--- a/setup.py
+++ b/setup.py
@@ -117,6 +117,9 @@ def build_isa_l():
     if SYSTEM_IS_UNIX:
         if platform.machine() == "aarch64":
             cflags_param = "CFLAGS_aarch64"
+        elif os.environ.get("OECORE_TARGET_ARCH") == "aarch64":
+            # Cross-compilation for aarch64
+            cflags_param = "CFLAGS_aarch64"
         else:
             cflags_param = "CFLAGS_"
         make_cmd = "make"
